src/imagery/i.eodag/i.eodag.py

```diff
@@ -710,6 +734,14 @@ def filter_result(search_result, geometry=None, queryables=None, **kwargs):
                     tmp_search_result_list.extend(filtered_search_result_list)
             search_result = SearchResult(tmp_search_result_list)
 
+    if options["pattern"]:
+        import re
+
+        pattern = re.compile(options["pattern"])
+        search_result = SearchResult(
+            filter(lambda p: pattern.fullmatch(p.properties["title"]), search_result)
```

Comment:
```suggestion
            filter(lambda p: pattern.fullmatch(p.properties["title"]), search_result)
```
It is suggested to use list comprehensions instead of lambda expression. See e.g.:
https://stackoverflow.com/questions/1632902/lambda-versus-list-comprehension-performance

Speaking of code-style. I reccomend checking your code with ruff, it will give you suggestions for how code-style could be improved. Checking code with ruff is quite instructive. I learned a lot from it. What you do is roughly the following:
```
python3 -m pip install ruff
ruff check --preview i.eodag.py
```

The point is not to do everything ruff suggests but to choose the suggestions that seem reasonable....

Reply:
> It is suggested to use list comprehensions instead of lambda expression. See e.g.: https://stackoverflow.com/questions/1632902/lambda-versus-list-comprehension-performance

On it.
> Speaking of code-style. I reccomend checking your code with ruff, it will give you suggestions for how code-style could be improved. Checking code with ruff is quite instructive. I learned a lot from it. What you do is roughly the following:
> 
> ```
> python3 -m pip install ruff
> ruff check --preview i.eodag.py
> ```
> 
> The point is not to do everything ruff suggests but to choose the suggestions that seem reasonable....

Nice thanks, it caught some typos/bugs.